api/client/src/main/java/org/projectnessie/client/builder/BaseOnReferenceBuilder.java

```diff
@@ -15,24 +15,54 @@
  */
 package org.projectnessie.client.builder;
 
-import org.projectnessie.client.api.OnReferenceBuilder;
+import org.projectnessie.model.Reference;
 
-abstract class BaseOnReferenceBuilder<R extends OnReferenceBuilder<R>>
-    implements OnReferenceBuilder<R> {
+public abstract class BaseOnReferenceBuilder<R> {
```

Comment:
I'd prefer to keep this class package-private, because users could simply up-cast to this one.
I suspect, this should still work with the added `public` functions.

Reply:
These builders are actually meant for reuse. Still you comment is relevant. I've reworked the interaces and base builders in the latest push. Now builders should be aligned to the java API, so casting to the will not expose extra / irrelevant functionality.... Is that ok from you POV?